integration-tests/tests/client/sharding_upgrade.rs

```diff
@@ -179,6 +179,55 @@ impl TestShardUpgradeEnv {
         }
     }
 
+    fn check_next_block_with_new_chunk(&mut self, height: BlockHeight) {
+        let block = self.env.clients[0].chain.get_block_by_height(height).unwrap().clone();
+        let block_hash = block.hash();
+        let num_shards = block.chunks().len();
+        for shard_id in 0..num_shards {
+            let mut last_block_hash_with_empty_chunk = match self.env.clients[0]
+                .chain
+                .get_next_block_hash_with_new_chunk(block_hash, shard_id as ShardId)
+                .unwrap()
+            {
+                Some((new_block_hash, target_shard_id)) => {
+                    let new_block =
+                        self.env.clients[0].chain.get_block(&new_block_hash).unwrap().clone();
+                    let chunks = new_block.chunks();
+                    // check that the target chunk in the new block is not empty
+                    assert_eq!(
+                        chunks.get(target_shard_id as usize).unwrap().height_included(),
+                        new_block.header().height(),
+                    );
+                    if chunks.len() == num_shards {
+                        assert_eq!(target_shard_id, shard_id as ShardId);
+                    }
+                    new_block.header().prev_hash().clone()
+                }
+                None => self.env.clients[0].chain.head().unwrap().last_block_hash.clone(),
+            };
+            // check that the target chunks in all prev blocks are empty
+            while &last_block_hash_with_empty_chunk != block_hash {
+                let last_block = self.env.clients[0]
+                    .chain
+                    .get_block(&last_block_hash_with_empty_chunk)
+                    .unwrap()
+                    .clone();
+                let chunks = last_block.chunks();
+                if chunks.len() == num_shards {
+                    assert_ne!(
+                        chunks.get(shard_id).unwrap().height_included(),
+                        last_block.header().height()
+                    );
+                } else {
+                    for chunk in chunks.iter() {
```

Comment:
this is a little bit unexpected (as you assume here that the block has no chunks at all, right?)

maybe add a comment

Reply:
A block actually always contain chunks for all shards in the chunks vector. When there is no new chunk for a shard, it simply copies the chunk for this shard from the last block. That's why I'm checking chunk.height_included() != last_block.header().height() here, that means the height that this chunk was included is not equal to the block height, which means the chunk is not included in this block.